semgrep-core/src/parsing/Test_parsing.ml

```diff
@@ -225,15 +227,23 @@ let parsing_common ?(verbose = true) lang xs =
     Lang.files_of_dirs_or_files lang xs
     |> Skip_code.filter_files_if_skip_list ~root:xs
     (* extra filtering excluding some extensions like '.d.ts'
-     * TODO? some source files are also really huge (> 20 MB), maybe
-     * we should skip them via Guess_lang.is_acceptable? or via another
-     * filter? and not count them in the statistics? They cause
-     * some annoying 'out of memory' errors that sometimes even the use
-     * of mem_limit_mb does not solve.
      * TODO: should become part of files_of_dirs_or_files
      *)
     |> List.filter (Guess_lang.is_acceptable lang)
+    (* Some source files are really huge (> 20 MB) and they cause
+     * some annoying 'out of memory' crash that sometimes even the use
+     * of mem_limit_mb above does not solve. Maybe it's because
+     * we run lots of parsing jobs in parallel in parsing-stats/run-all
+     * TODO: we should do the parallel work OCaml side, not in run-all.
+     * TODO: improve the engine so even those huge files do not cause errors
+     * TODO: should we skip them via Guess_lang.is_acceptable?
+     *)
+    |> Common.exclude (fun file ->
+           let res = Common2.filesize file > 5_000_000 in
+           if res then pr2 (spf "skipping %s, too big" file);
+           res)
```

Comment:
I think we should reserve the right to skip any file that looks like it was machine-generated. This would automatically exclude very large files. I already tried a few things when working on spacegrep. I'll look into this again.